design/design-generic-references.md

```diff
@@ -0,0 +1,477 @@
+# Generic Resource References
+
+* Owner: Predrag Knezevic (@pedjak)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+Crossplane providers provision and monitor external resources in third party
+cloud providers, following the declared specification in the corresponding
+managed resources. Very often, creation of these objects cannot happen in
+parallel, i.e. some fields can be set only after a given dependency is created.
+
+[Cross resource referencing] solves this problem for majority of cases by
+following [Kubernetes API object reference] convention:
+
+```yaml
+spec:
+  fooRef:
+    name: bar
+```
+
+Field `spec.fooRef` contains the name of the resource where the value
+for `spec.foo` is to be found. The kind of referred resource and the path to the
+field containing the value used later for setting `spec.foo` is fixed, i.e. it
+is embedded in the resolving mechanism used by the provider.
+
+Although the used convention covers the majority of use cases, user experience
+shows that there is still a need to refer a value on an arbitrary path within an
+object of arbitrary kind. For example, 
+[`datasync.aws.upbound.io` `Task`](https://github.com/upbound/provider-aws/blob/main/apis/datasync/v1beta1/zz_task_types.go#L375)
+has [`spec.forProvider.sourceLocationArn`](https://github.com/upbound/provider-aws/blob/main/apis/datasync/v1beta1/zz_task_types.go#L359)
+field that can obtain value from an instance of the following object types:
+
+* LocationS3
+* LocationSmb
+* LocationObjectStorage
+* LocationNfs
+* LocationHdfs
+* LocationFsxWindowsFileSystem
+* LocationFsxOpenzfsFileSytem
+* LocationFsxLustreFileSystem
+* LocationEfs
+
+However, [the implemented provider resolving mechanism](https://github.com/upbound/provider-aws/blob/main/apis/datasync/v1beta1/zz_generated.resolvers.go#L104-L112)
+looks only for the value in `LocationS3` instance named
+by `spec.forProvider.sourceLocationArnRef` field.
+
+Currently, the issue can be solved by defining the following composition:
+
+* Declare under managed resources the needed source object (one of `Location*`)
+  types with [observe only management policy]
+* Patch composite object with the value from the source object
+* Patch Task's `spec.forProvider.sourceLocationArn` field with the composite
+  value
+
+The proposed solution has the following drawbacks:
```

Comment:
Making sure I understand: by "the proposed solution" do you mean the pattern you describe above that requires using a Composition?

Reply:
I would rephrase it as "this solution" so it's clear what it is referring to.